### DISCUSSION TITLE
Forward Erasure Correction Codes

### DISCUSSION THREAD

**Discussion by DanOpoka** [yamcs/yamcs#701] [Q&A] [answered]
Hello,

I have been reading the:
-  **CCSDS - ERASURE CORRECTING CODES FOR USE IN NEAR-EARTH AND DEEP-SPACE COMMUNICATIONS**
As Yamcs follows the CCSDS, I've been wondering the following:

Are **Forward Erasure Correcting Codes** already implemented into Yamcs and where can they be found?
If not, where could they be added?

Any help would be greatly appreciated.
Thanks.

**Comment by cpbridges** [comment, marked as answer]
I hope people will correct me if I'm wrong but yamsc is a mission control system - any pipeline for your bit level communications would be in addition to this. I would recommend looking at 'gnuradio' and 'gr_satellites'.

**Reply by xpromache** [reply]
We had a discussion in private with Daniel - the kind of correction codes specified in CCSDS 131.5-O-1 work at higher level - they work at the level of frames (or CFDP PDUs) rather than bits. Of course they could be implemented in gnuradio but they could also be implemented in software since they requires accumulation of larger amount of data than the usual error correction codes.

In any case, the answer is that currently Yamcs does not support the CCSDS Forward Erasure Correcting Codes.
